app/src/main/java/de/danoeh/antennapod/ui/episodeslist/EpisodeItemListAdapter.java

```diff
@@ -229,6 +229,16 @@ public boolean onContextItemSelected(MenuItem item) {
         return false;
     }
 
+    public List<FeedItem> getPlayedItems() {
```

Comment:
I think I would rather let the database do the filtering. That way we de-couple UI and logic a bit more. Something like this should work (from the fragment, not the adapter):

```java
DBReader.getEpisodes(0, Integer.MAX_VALUE, new FeedItemFilter(
        FeedItemFilter.DOWNLOADED, FeedItemFilter.INCLUDE_NOT_SUBSCRIBED, FeedItemFilter.PLAYED), 
        SortOrder.DATE_OLD_NEW)
```

Reply:
Thanks for the tip